docs/hpc/08_ood/open_on_demand.md

```diff
@@ -0,0 +1,184 @@
+
+Next, activate your conda environment with the following:
+```
+source /ext3/env.sh
+```
+
+### Install Packages to Miniforge Environment
+Now that your environment is activated, you can update and install packages
+```
+conda config --remove channels defaults
+conda update -n base conda -y
+conda clean --all --yes
+conda install pip --yes
+conda install ipykernel --yes # Note: ipykernel is required to run as a kernel in the Open OnDemand Jupyter Notebooks
+```
+To confirm that your environment is appropriately referencing your Miniforge installation, try out the following:
+```
+unset which
+which conda
+# output: /ext3/miniforge3/bin/conda
+
+which python
+# output: /ext3/miniforge3/bin/python
+
+python --version
+# output: Python 3.8.5
+
+which pip
+# output: /ext3/miniforge3/bin/pip
+```
+
+Now use either conda install or pip to install your required python packages to the Miniforge environment.
+
+To install larger packages, like Tensorflow, you must first start an interactive job with adequate compute and memory resources to install packages. The login nodes restrict memory to 2GB per user, which may cause some large packages to crash.
+```
+srun --cpus-per-task=2 --mem=10GB --time=04:00:00 --pty /bin/bash
+
+# wait to be assigned a node
+
+singularity exec --overlay /scratch/$USER/my_env/overlay-15GB-500K.ext3:rw /scratch/work/public/singularity/cuda12.3.2-cudnn9.0.0-ubuntu-22.04.4.sif /bin/bash
+
+source /ext3/env.sh
+# activate the environment
+```
+
+After it is running, you’ll be redirected to a compute node. From there, run singularity to setup on conda environment, same as you were doing on login node.
+
+### Configure iPython Kernels
+To create a kernel named my_env copy the template files to your home directory.
+```
+mkdir -p ~/.local/share/jupyter/kernels
+cd ~/.local/share/jupyter/kernels
+cp -R /share/apps/mypy/src/kernel_template ./my_env # this should be the name of your Singularity env
+cd ./my_env 
+
+ls
+#kernel.json  logo-32x32.png  logo-64x64.png  python # files in the ~/.local/share/jupyter/kernels directory
+```
+
+To set the conda environment, edit the file named 'python' in /.local/share/jupyter/kernels/my_env/. 
+
+The python file is a wrapper script that the Jupyter notebook will use to launch your Singularity container and attach it to the notebook.
+
+At the bottom of the file we have the template singularity command.
+```
+singularity exec $nv \
+  --overlay /scratch/$USER/my_env/overlay-15GB-500K.ext3:ro \
+  /scratch/work/public/singularity/cuda12.3.2-cudnn9.0.0-ubuntu-22.04.4.sif \
+  /bin/bash -c "source /ext3/env.sh; $cmd $args"
+```
+***WARNING:*** If you used a different overlay (/scratch/$USER/my_env/overlay-15GB-500K.ext3 shown above) or .sif file (/scratch/work/public/singularity/cuda12.3.2-cudnn9.0.0-ubuntu-22.04.4.sif shown above), you MUST change those lines in the command above to the files you used.
```

Comment:
I'd prefer to use admonitions https://docusaurus.io/docs/markdown-features/admonitions

Reply:
Oh, yeah, that is better.  Thanks for catching it.